### PR TITLE
Replace Logger with Log

### PR DESCRIPTION
**Pull request by drujensen**
### Description of the Change

Replace Logger with Log to remove deprecated warning

To eliminate all deprecation warnings, Granite and Quartz Mailer will need to be released and shards template updated.

### Benefits

Removes the deprecated warning.

### CODE REVIEW

File: src/amber/cli/commands/pipelines.cr [hunk 1]
```diff
@@ -77,6 +77,7 @@ module Amber::CLI
           case line
           when .starts_with?("pipeline") then set_pipe(line)
           when .starts_with?("plug")     then set_plug(line)
+          else
```
Comment (robacarp) [inline src/amber/cli/commands/pipelines.cr:80]: Should this and [other](https://github.com/amberframework/amber/pull/1201/files#diff-eb22ae110d5e9148a4e2657f50faf2e7R85) places where Crystal wants an else have a puts, raise, or something else so they don't fail silently?

Comment (drujensen) [reply src/amber/cli/commands/pipelines.cr:80]: good idea

File: src/amber/environment/logging.cr [hunk 2]
```diff
@@ -3,12 +3,12 @@ module Amber::Environment
     alias OptionsType = Hash(String, String | Bool | Array(String))
 
     SEVERITY_MAP = {
-      "debug":   Logger::DEBUG,
-      "info":    Logger::INFO,
-      "warn":    Logger::WARN,
-      "error":   Logger::ERROR,
-      "fatal":   Logger::FATAL,
-      "unknown": Logger::UNKNOWN,
+      "debug":   Log::Severity::Debug,
```
Comment (eliasjpr) [inline src/amber/environment/logging.cr:6]: We can delete this hash. See below.

File: src/amber/environment/logging.cr [hunk 3]
```diff
@@ -57,7 +57,7 @@ module Amber::Environment
       @context = logging["context"].as(Array(String))
     end
 
-    def severity : Logger::Severity
+    def severity : Log::Severity
       SEVERITY_MAP[@severity]
```
Comment (eliasjpr) [inline src/amber/environment/logging.cr:61]: This can be replace with `Logger::Severity.parse @severity` allowing to remove the map

**Comment by drujensen** [comment]
@eliasjpr I have been considering ditching the whole `Logger` and replace that functionality with `Log = Log.for(self)` throughout the whole code base.  This will give the end user much more control on what they want to set a `debug` vs `warn`, etc.  wdyt?

**Comment by drujensen** [comment]
I'm also struggling to figure out why the tests are failing.  Seems unrelated to the changes I'm making.

**Comment by eliasjpr** [comment]
@drujensen I agree to replace the current implementation with the new Log approach. If you would like we can discuss in a hangout call with the team.

**Comment by drujensen** [comment]
@eliasjpr Thanks. I think I have an idea on how to take advantage of `Log` without losing some of the work we did in the Logger Pipeline.  I will let you know.

I finally have passing specs.  I usually use docker locally to run specs but crystal is failing and locking up on `Log.write` to io.  After Ctrl+C, it spits out a bunch of locked io write failures.

I find this a little worrisome if the new `Log` doesn't work in Docker:
```
....................................................................................................................................................E............................................................................................................EEEEEEEEEEEEEE.puts([:a] + [:b])..result = (
  1337
)
puts result.inspect..EEEEEEEEEEEEE................EEEE......Unhandled exception in spawn: Closed stream (IO::Error)
  from /usr/share/crystal/src/io.cr:119:5 in 'check_open'
  from /usr/share/crystal/src/io/buffered.cr:131:15 in 'write'
  from /usr/share/crystal/src/io.cr:472:7 in 'write_utf8'
  from /usr/share/crystal/src/string.cr:4413:5 in 'to_s'
  from /usr/share/crystal/src/io.cr:175:5 in '<<'
  from src/amber/environment/logger.cr:29:9 in '->'
  from /usr/share/crystal/src/log/io_backend.cr:255:3 in 'format'
  from /usr/share/crystal/src/log/io_backend.cr:14:7 in 'write'
  from src/amber/environment/logger.cr:36:3 in 'info'
  from src/amber/environment/logger.cr:38:5 in 'info'
  from src/amber/websockets/server.cr:7:9 in 'create_endpoint'
  from spec/support/helpers/websockets_helper.cr:14:7 in '->'
  from /usr/share/crystal/src/fiber.cr:255:3 in 'run'
  from /usr/share/crystal/src/fiber.cr:92:34 in '->'
  from ???
```

File: src/amber/cli/commands/pipelines.cr (hunk 1 above)
Comment (drujensen) [reply src/amber/cli/commands/pipelines.cr:81]: This one actually needs to skip lines that are not pipeline so its not a failure case.

File: src/amber/cli/templates/app/config/environments/development.yml.ecr [hunk 4]
```diff
@@ -25,10 +25,11 @@ redis_url: "redis://localhost:6379"
 when "mysql" -%>
 database_url: mysql://root@localhost:3306/<%= database_name %>_development
 <% when "pg" -%>
-database_url: postgres://postgres:@localhost:5432/<%= database_name %>_development
+database_url: postgres://postgres:password@localhost:5432/<%= database_name %>_development
 <% when "sqlite" -%>
 database_url: sqlite3:./db/<%= database_name %>_development.db
-<% end -%>
+<% else
```
Comment (drujensen) [inline src/amber/cli/templates/app/config/environments/development.yml.ecr:31]: Not setting a database_url makes sense in this case.

**Comment by drujensen** [comment]
@eliasjpr @robacarp Lets get this one merged.  I will look into fully leveraging the `Log` capabilities in a future release.

**Pull request merged**